### PR TITLE
REF: re-raise AssertionError unchanged

### PR DESCRIPTION
**Pull request by jbrockmendel**
This should be all the remaining places where we catch `Exception` in groupby/apply/resample code.  This should make debugging much easier going forward.

cc @jreback @WillAyd

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -251,6 +251,8 @@ def aggregate(self, func=None, *args, **kwargs):
 
             try:
                 return self._python_agg_general(func, *args, **kwargs)
+            except AssertionError:
```
Comment (WillAyd) [inline pandas/core/groupby/generic.py]: Hmm what's the point of the bare raise here? I get needed to differentiate now from the `except Exception`, but coupled with change above is there something actually throwing the `AssertionError`? Or do we mean to be catching `ValueError` here?

Comment (jbrockmendel) [reply pandas/core/groupby/generic.py]: > but coupled with change above is there something actually throwing the AssertionError

No, which means that during development we can now add assertions and not have them get swallowed.

**Comment by jreback** [comment]
lgtm. pls rebase

**Comment by jbrockmendel** [comment]
rebased+green

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**